### PR TITLE
[Inference] Optimize the writing style of tail processing and the logic related to macro definitions.

### PR DESCRIPTION
**Pull request by yuehuayingxueluo**
…efinitions have been optimized.

## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.
pytest:
image

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: extensions/csrc/cuda/context_kv_cache_memcpy_kernel.cu [hunk 1]
```diff
@@ -93,76 +95,61 @@ void apply_context_kv_cache_memcpy(
...
+                batch_size,                                                                             \
+                block_table_stride,                                                                     \
+                key_stride,                                                                             \
+                value_stride                                                                            \
+            );                                                                                          \
+    } while(0)
+
+#define CONTEXT_KV_CACHE_MEMCOPY_KERNEL_LAUNCH_VEC_SIZE_CASE(__aligned)                                 \
+    do {                                                                                                \
+        switch (vec_size) {                                                                             \
```
Comment (yuanheng-zhao) [inline extensions/csrc/cuda/context_kv_cache_memcpy_kernel.cu:131]: Where does `vec_size` come from

Comment (yuehuayingxueluo) [reply extensions/csrc/cuda/context_kv_cache_memcpy_kernel.cu:131]: image
The vec_size is defined here.

Comment (yuanheng-zhao) [reply extensions/csrc/cuda/context_kv_cache_memcpy_kernel.cu:131]: Got it.

**Review by Courtesy-Xs — APPROVED** [review]
LGTM

**Pull request merged**